rfd/0002-custom-terraform-registry.md

```diff
@@ -0,0 +1,185 @@
+---
+authors: Trent Clarke (trent@goteleport.com)
+state: draft
+---
+
+# RFD 2 - Custom Terraform Registry
+
+## What
+
+A Terraform _registry_ is an index of available terraform _providers_. The
+Terraform tool uses registries to locate, validate and install Terraform
+providers for the various services required by terraform scripts. Hashicorp
+(the Terraform vendor) publishes the protocol that Terraform uses to interact
+with registries and so third parties (like us) may provide _custom_ Terraform
+registries outside the main, Hashicorp-provided public registry.
+
+This RFD investigates the use of a custom Terraform registry to distribute the
+Teleport provider for Terraform.
+
+## Why
+
+We have two main goals that effect how we offer our Terraform provider to
+customers
+
+ 1. We want to meet out customers where they are, e.g. if they're using
+    Terraform, we want to supply them with a Terraform provider, and a method
+    for installing it that is natural to the Terraform ecosystem.
+
+ 2. We want to maintain close control of the distribution of our software.
+
+With these two goals in mind, the most practical solution for distributing
+the Teleport Terraform provider appears to be hosting a custom Terraform
+registry.
+
+### Why not use the public Hashicorp registry?
```

Comment:
You may want / need to check in with potential customers who are looking for this on whether they would accept this or not. I got from @FrankWeissmann on a side discussion for something else today, and this came up, that a customer is really pushing for this because they have a policy of not using any third party plugins, so they might see this in a similar light.

Reply:
Having some discussions with @FrankWeissmann and the client involved now. Hopefully we can come to a resolution that works for everyone.